modules/react/tabs/lib/cursor/useRovingFocus.tsx

```diff
@@ -67,10 +77,11 @@ export const useRovingFocus = createHook(
     return {
       onKeyDown(event: React.KeyboardEvent) {
         (Object.keys(
-          orientationKeyMap[state.orientation]
+          state.columnCount > 0 ? gridKeyMap : orientationKeyMap[state.orientation]
         ) as (keyof typeof orientationKeyMap[typeof state.orientation])[]).forEach(key => {
           if (isRTL ? event.key === rightToLeftMap[key] : event.key === key) {
-            const event = orientationKeyMap[state.orientation][key];
+            const event =
+              state.columnCount > 0 ? gridKeyMap[key] : orientationKeyMap[state.orientation][key];
```

Comment:
Conditionals to check for column count, if it's greater than 0, we assume it has some column and therefore it's a grid, changing the mapping of keys